.werft/jobs/build/deploy-to-preview-environment.ts

```diff
@@ -521,7 +521,7 @@ export async function issueMetaCerts(
     metaClusterCertParams.ip = getCoreDevIngressIP();
     metaClusterCertParams.bucketPrefixTail = "";
     metaClusterCertParams.additionalSubdomains = additionalSubdomains;
-    await issueCertificate(werft, metaClusterCertParams, { ...metaEnv(), slice });
+    return await issueCertificate(werft, metaClusterCertParams, { ...metaEnv(), slice });
```

Comment:
Do I understand correctly that the `await` here forces the process to wait for the whole execution of `issueCertificate` before continuing?

Isn't it the opposite of what you want to achieve?

Reply:
My knowledge of TypeScript's syntax might be failing me. I'm treating running an async function in a similar way to go routines or Rust's `async` (although it seems like sync and async are not mutually exclusive in TS).

As this is wrapped by an async function (`var certReady = issueCertificate(werft, config);`), which we don't await on, it doesn't matter what goes on inside (i.e. we basically treat this synchronously). From my understanding, this means this function runs in the background, whether we have an `await` internal to the function or not.

I could remove the awaits and pass the promise from the downstream functions, but, from my understanding, it wouldn't impact this function

Reply:
Yes, this was my understanding. As there is no plan to add further concurrency into the function it makes sense to just return the promises upstream, rather than awaiting here